packages/client/src/sync/fetcher/trienodefetcher.ts

```diff
@@ -365,9 +365,7 @@ export class TrieNodeFetcher extends Fetcher<JobTask, Uint8Array[], Uint8Array>
               for (const [path, data] of pathToStorageNode) {
                 const storageNode = decodeMPTNode(data)
                 if (storageNode instanceof LeafMPTNode) {
-                  const storageKey = bytesToUnprefixedHex(
-                    pathToHexKey(path, storageNode.key(), 'keybyte'),
-                  )
+                  const storageKey = bytesToHex(pathToHexKey(path, storageNode.key(), 'keybyte'))
```

Comment:
We should keep using unprefixedHex here as optimization.

Reply:
(pending because this likely should be changed because of bugs in this file (?))

Reply:
Same thing as above where the code as-is would throw (I would then assume that this is untested? Otherwise we would've noticed).

Reply:
I have forwarded this, we can merge this as-is from my side but it seems like this needs some eyes :eyes: (in a different PR)